src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActions/SuggestedAction.cs

```diff
@@ -141,39 +141,55 @@ private async Task InvokeWorkerAsync(IProgressTracker progressTracker, Cancellat
         {
             await this.ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
 
-            IEnumerable<CodeActionOperation> operations = null;
+            object options = null;
             if (CodeAction is CodeActionWithOptions actionWithOptions)
             {
-                var options = actionWithOptions.GetOptions(cancellationToken);
-                if (options != null)
-                    operations = await GetOperationsAsync(actionWithOptions, options, cancellationToken).ConfigureAwait(true);
-            }
-            else
-            {
-                operations = await GetOperationsAsync(progressTracker, cancellationToken).ConfigureAwait(true);
+                using (Logger.LogBlock(FunctionId.CodeFixes_GetOptions, KeyValueLogMessage.Create(LogType.UserAction, m => CreateLogProperties(m)), cancellationToken))
+                {
+                    options = actionWithOptions.GetOptions(cancellationToken);
+                }
             }
 
-            AssertIsForeground();
-
-            if (operations != null)
+            using (Logger.LogBlock(
+                   FunctionId.CodeFixes_GetAndApplyChanges, KeyValueLogMessage.Create(LogType.UserAction, m => CreateLogProperties(m)), cancellationToken))
```

Comment:
this is named "GetAndApplyChanges" but doedsn't apply the changes.

Reply:
```
                        await EditHandler.ApplyAsync(
                            Workspace,
                            OriginalSolution,
                            document,
                            operations.ToImmutableArray(),
                            CodeAction.Title,
                            progressTracker,
                            cancellationToken).ConfigureAwait(false);
```
Does this part track the cost to apply the change?